src/common/types/timestamp.cpp

```diff
@@ -298,44 +298,62 @@ timestamp_t Timestamp::GetCurrentTimestamp() {
 
 timestamp_t Timestamp::FromEpochSeconds(int64_t sec) {
 	int64_t result;
+	D_ASSERT(Timestamp::IsFinite(timestamp_t(sec)));
 	if (!TryMultiplyOperator::Operation(sec, Interval::MICROS_PER_SEC, result)) {
 		throw ConversionException("Could not convert Timestamp(S) to Timestamp(US)");
 	}
 	return timestamp_t(result);
 }
 
-timestamp_t Timestamp::FromEpochMs(int64_t ms) {
+timestamp_t Timestamp::FromEpochMsPossiblyInfinite(int64_t ms) {
```

Comment:
Does this need to be publicly visible?

Reply:
It's used in `date_part.cpp` `EpochMillisOperator::Inverse` so yes I think so.

PossiblyInfinite just means that it won't return if `ms` is equal to `timestamp::infinity()`, and instead it just throws when trying to multiply if that's the case